tests/io/test_processor.py

```diff
@@ -33,3 +109,66 @@ def test_processor_invoke_no_records(reset_handler, mocker, caplog):
     handler(event, None)
 
     assert "Successfully processed 0 records" in caplog.messages
+
+
+def test_processor_kinesis_dynamodb_insert_update(cratedb, reset_handler, mocker, caplog):
+    """
+    Roughly verify that the unified Lambda handler works with AWS DynamoDB.
+    """
+
+    # Define target table name.
+    table_name = '"testdrive"."demo"'
+
+    # Create target table.
+    cratedb.database.run_sql(DynamoDBCDCTranslator(table_name=table_name).sql_ddl)
+
+    # Configure Lambda processor per environment variables.
+    handler_environment = {
+        "CRATEDB_SQLALCHEMY_URL": cratedb.get_connection_url(),
+        "MESSAGE_FORMAT": "dynamodb",
+        "CRATEDB_TABLE": table_name,
+    }
+    mocker.patch.dict(os.environ, handler_environment)
+
+    from cratedb_toolkit.io.processor.kinesis_lambda import handler
+
+    # Define two CDC events: INSERT and UPDATE.
+    # They have to be conveyed separately because CrateDB needs a
+    # `REFRESH TABLE` operation between them.
+    event1 = {
+        "Records": [
+            wrap_kinesis(DYNAMODB_CDC_INSERT_NESTED),
+        ]
+    }
+    event2 = {
+        "Records": [
+            wrap_kinesis(DYNAMODB_CDC_MODIFY_NESTED),
+        ]
+    }
+
+    # Run transfer command.
+    handler(event1, None)
+    cratedb.database.refresh_table(table_name)
+    handler(event2, None)
+    cratedb.database.refresh_table(table_name)
```

Comment:
By adding and exercising this test case, I observed that an INSERT operation followed by an UPDATE operation on the same document/record indeed requires a REFRESH operation in between.

Reply:
This section has been amended in favor of adding the REFRESH TABLE operation to the handler.